src/spootnik/reporter/impl.clj

```diff
@@ -241,42 +265,99 @@
     (if started?
       this
-      (let [prometheus-registry (CollectorRegistry/defaultRegistry)
-            rclient             (when riemann (riemann-client riemann))
-            [reg reps]          (build-metrics metrics rclient prometheus-registry)
-            options             (when sentry (or raven-options {}))
-            prometheus-server   (when prometheus
-                                  (let [tls (:tls prometheus)
-                                        opts (cond->
-                                                 {:port (:port prometheus)}
-                                               (some? (:tls prometheus))
-                                               (assoc :ssl-context (ssl-context tls))
-                                               (:host prometheus)
-                                               (assoc :socket-address
-                                                      (InetSocketAddress.
-                                                       (:host prometheus)
-                                                       (:port prometheus))))]
-                                    (http/start-server
-                                     (partial prometheus-handler
-                                              prometheus
-                                              prometheus-registry)
-                                     opts)))]
+      (let [prometheus-registry  (CollectorRegistry/defaultRegistry)
+            pushgateway-job      (:job pushgateway)
+            pushgateway-client   (build-pushgateway-client pushgateway)
+            pushgateway-registry (CollectorRegistry.)
+            pushgateway-metrics  (into {} (map #(vector (:name %1) (-> (build-pushgateway-collector %1)
+                                                                       (register-pushgateway-collector pushgateway-registry)))
+                                               (:metrics pushgateway)))
+            rclient              (when riemann (riemann-client riemann))
+            [reg reps]           (build-metrics metrics rclient prometheus-registry pushgateway-registry)
+            options              (when sentry (or raven-options {}))
+            prometheus-server    (when prometheus
+                                   (let [tls (:tls prometheus)
+                                         opts (cond->
+                                               {:port (:port prometheus)}
+                                                (some? (:tls prometheus))
+                                                (assoc :ssl-context (server-ssl-context tls))
+                                                (:host prometheus)
+                                                (assoc :socket-address
+                                                       (InetSocketAddress.
+                                                        (:host prometheus)
+                                                        (:port prometheus))))]
```

Comment:
```suggestion
                                                        ^int (:port prometheus))))]
```

Reply:
Fixed in 13ab7542552e4822cff49cb2ed11926cca2a5eab :